string max_min
int8 diff
